haskell-lsp-types/src/Language/Haskell/LSP/Types/Uri.hs

```diff
@@ -26,8 +27,10 @@ newtype NormalizedUri = NormalizedUri Text
   deriving (Eq,Ord,Read,Show,Generic,Hashable)
 
 toNormalizedUri :: Uri -> NormalizedUri
-toNormalizedUri (Uri t) =
+toNormalizedUri uri =
     NormalizedUri $ T.pack $ escapeURIString isUnescapedInURI $ unEscapeString $ T.unpack t
+  where (Uri t) = maybe uri filePathToUri (uriToFilePath uri)
```

Comment:
I am not sure if it is the best way to do it.
`filepathToUri` main work is to prepare `FilePath` to convert it to `Uri` (except making the driver letter upper case)